// react-recaptcha-v2/ReCaptcha.js
import React from 'react'
import PropTypes from 'prop-types'
import getReCaptchaScript from './getReCaptchaScript'
import ReCaptchaCore from './ReCaptchaCore'
import reCaptchaId from './reCaptchaId'

let reCaptchaScript = null

async function loadScript (locale) {
  reCaptchaScript = await getReCaptchaScript(locale)
}

export default class ReCaptcha extends React.Component {
  static reset (name) {
		if (!reCaptchaScript || reCaptchaId[name] === undefined) return
    reCaptchaScript.reset(reCaptchaId[name])
  }

  static async updateLocale (locale) {
    await loadScript()
    Object.entries(reCaptchaId).forEach(([name, id]) => {
      recaptchaScript.reset(id)
    })
  }

  static propTypes = {
    name: PropTypes.string,
    locale: PropTypes.string,
  }

  state = {
    isReady: false,
  }

  async componentDidMount () {
    const { locale = "en" } = this.props
    if (!reCaptchaScript) {
      await loadScript()
    }
    this.setState({ isReady: true })
  }

  render () {
    const { isReady } = this.state

    if (!isReady) return null
    return <ReCaptchaCore reCaptchaScript={reCaptchaScript} {...this.props} />
  }
}
