// react-recaptcha-v2/ReCaptchaCore.js
import React, { Component, createRef } from 'react'
import PropTypes from 'prop-types'
import reCaptchaId from './reCaptchaId'

export default class ReCaptchaCore extends Component {
  static propTypes = {
    onChange: PropTypes.func,
    reCaptchaScript: PropTypes.object,
    theme: PropTypes.oneOf(['dark', 'light']),
    size: PropTypes.oneOf(['normal', 'compact']),
    tabindex: PropTypes.number,
    name: PropTypes.string,
    className: PropTypes.string,
    siteKey: PropTypes.string.isRequiered,
  }

  componentDidMount () {
    const { siteKey, name, onChange, theme, size, tabindex, reCaptchaScript } = this.props
    reCaptchaId[name] = reCaptchaScript.render(this.reCaptchaRef.current, {
      sitekey,
      callback: onChange,
      theme,
      size,
      tabindex,
      'expired-callback': this.onExpired,
    })
  }

  onExpired = () => {
    this.props.onChange && this.props.onChange(null)
  }

  reCaptchaRef = createRef()

  render () {
    const { className } = this.props
    return <div className={className} ref={this.reCaptchaRef} />
  }
}
